### PR TITLE
ddl: pass runtime info through args for reorganize partition related

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #54436 

Problem Summary:

### Check List

Tests 

- [x] Unit test, existing cases should cover it
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/partition.go [hunk 1]
```diff
@@ -3127,16 +3119,22 @@ func getReorgPartitionInfo(t *meta.Mutator, job *model.Job) (*model.TableInfo, [
...
+	args, err := model.GetTablePartitionArgs(job)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+	jobCtx.jobArgs = args
 	// Handle the rolling back job
 	if job.IsRollingback() {
-		ver, err := w.rollbackLikeDropPartition(jobCtx, job)
+		ver, err := w.rollbackLikeDropPartition(jobCtx, job, args)
```
Comment (joechenrh) [inline pkg/ddl/partition.go:3130]: No need to pass `args`?

Comment (D3Hunter) [reply pkg/ddl/partition.go]: indeed https://github.com/pingcap/tidb/pull/56656/commits/bb592f81ef59dddb9a518b1253297b503631f6ff

**Review by joechenrh — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**